.github/workflows/cargo-deny.yml

```diff
@@ -0,0 +1,42 @@
+name: cargo deny checks
+
+on:
+  pull_request:
+    paths:
+      - 'Cargo.lock'
+  schedule:
+    - cron: '0 0 * * *'
+
+jobs:
+  cargo-deny:
+    strategy:
+      matrix: >-
+        ${{
+          fromJSON(
+            github.event_name == 'schedule'
+              && '{"ref":["main","release","release-proxy","release-compute"]}'
+              || format('{{"ref":["{0}"]}}', github.sha)
+          )
+        }}
```

Comment:
Looks nice!
I would probable move `ref` from an object in expression to raw yaml, to make it easier to read i.e.:
```
matrix:
  ref: >-
    ${{ ... }}
```

Reply:
good point, done.